### PR TITLE
Keys.ts fully documented

### PR DESCRIPTION
**Pull request by dylanireland**

### CODE REVIEW

**Comment by hoffmannjan** [comment]
@dylanireland is it ready to review? :)

**Comment by dylanireland** [comment]
@hoffmannjan It's ready! Thanks

**Comment by hoffmannjan** [comment]
Sorry for the delays, hope to have time to review this next sprint... Cheers @dylanireland

File: src/lib/Keys.ts [hunk 1]
```diff
@@ -391,13 +403,25 @@ export class Ed25519 extends AsymmetricKey {
   }
 }
 
+/**
+ * Secp256k1 variant of `AsymmetricKey`
+ * @privateRemarks
+ * Needs origin notation
```
Comment (gyroflaw) [inline src/lib/Keys.ts:409]: Which libraries can be the origin notation?
- [secp256k1](https://www.npmjs.com/package/secp256k1)
- [eccrypto](https://www.npmjs.com/package/eccrypto)
- [ethereum-cryptography/secp256k1](https://www.npmjs.com/package/ethereum-cryptography#secp256k1-curve)

FYI: @hoffmannjan

**Comment by gyroflaw** [comment]
I am not sure this is the right place to post this.
While reviewing this PR, I found that we're using [eccrypto](https://www.npmjs.com/package/eccrypto) module to generate `secp256k1` keypair even we can generate `secp256k1` keypair using `ethereum-cryptography/secp256k1` module.

Suggested change:

from
```ts
  public static new() {
    const privateKey = eccrypto.generatePrivate();
    const publicKey = Uint8Array.from(eccrypto.getPublicCompressed(privateKey));
    return new Secp256K1(publicKey, privateKey);
  }
```

to 
```ts
  public static new() {
    const privateKey = secp256k1.createPrivateKeySync();
    const publicKey = secp256k1.publicKeyCreate(privateKey);
    return new Secp256K1(publicKey, privateKey);
  }
```

**Review by gyroflaw — CHANGES_REQUESTED** [review]

**Comment by hoffmannjan** [comment]
> I am not sure this is the right place to post this. While reviewing this PR, I found that we're using [eccrypto](https://www.npmjs.com/package/eccrypto) module to generate `secp256k1` keypair even we can generate `secp256k1` keypair using `ethereum-cryptography/secp256k1` module.
> 
> Suggested change:
> 
> from
> 
> ```ts
>   public static new() {
>     const privateKey = eccrypto.generatePrivate();
>     const publicKey = Uint8Array.from(eccrypto.getPublicCompressed(privateKey));
>     return new Secp256K1(publicKey, privateKey);
>   }
> ```
> 
> to
> 
> ```ts
>   public static new() {
>     const privateKey = secp256k1.createPrivateKeySync();
>     const publicKey = secp256k1.publicKeyCreate(privateKey);
>     return new Secp256K1(publicKey, privateKey);
>   }
> ```

So this is a bigger change that we need to discuss. Please create an issue describing this - we will need to check and discuss if it doesn't break anything and if it works both in nodejs and browser.

**Comment by gyroflaw** [comment]
> > I am not sure this is the right place to post this. While reviewing this PR, I found that we're using [eccrypto](https://www.npmjs.com/package/eccrypto) module to generate `secp256k1` keypair even we can generate `secp256k1` keypair using `ethereum-cryptography/secp256k1` module.
> > Suggested change:
> > from
> > ```ts
> >   public static new() {
> >     const privateKey = eccrypto.generatePrivate();
> >     const publicKey = Uint8Array.from(eccrypto.getPublicCompressed(privateKey));
> >     return new Secp256K1(publicKey, privateKey);
> >   }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to
> > ```ts
> >   public static new() {
> >     const privateKey = secp256k1.createPrivateKeySync();
> >     const publicKey = secp256k1.publicKeyCreate(privateKey);
> >     return new Secp256K1(publicKey, privateKey);
> >   }
> > ```
> 
> So this is a bigger change that we need to discuss. Please create an issue describing this - we will need to check and discuss if it doesn't break anything and if it works both in nodejs and browser.

Sure will do 👍

File: src/lib/Keys.ts (hunk 1 above)
Comment (hoffmannjan) [reply src/lib/Keys.ts]: OK so we need our own notation, @gyroflaw could you prepare it and add as part of this PR?

Comment (gyroflaw) [reply src/lib/Keys.ts:409]: sure will do

Comment (hoffmannjan) [reply src/lib/Keys.ts]: nice!

**Review by gyroflaw — APPROVED** [review]

**Review by hoffmannjan — APPROVED** [review]

**Pull request merged**